src/hotspot/share/opto/parseHelper.cpp

```diff
@@ -358,6 +358,13 @@ void PEAState::add_new_allocation(Node* obj) {
 
   if (nfields >= 0) {
     AllocateNode* alloc = obj->in(1)->in(0)->as_Allocate();
+#ifndef PRODUCT
+    if (PEA_debug_idx > 0 && alloc->_idx != PEA_debug_idx) {         // only allow PEA_debug_idx
+        return;
```

Comment:
NIT: inconsistent indent.

Reply:
I see. I will update it after I confirm that it can compile on x86-32.